src/symbolize/gimli/parse_running_mmaps_unix.rs

```diff
@@ -0,0 +1,146 @@
+use super::mystd::io::BufRead;
+use super::{OsString, Vec};
+
+#[derive(PartialEq, Eq, Debug)]
+pub(super) struct MapsEntry {
+    /// start (inclusive) and limit (exclusive) of address range.
+    address: (usize, usize),
+    /// The perms field are the permissions for the entry
+    ///
+    /// r = read
+    /// w = write
+    /// x = execute
+    /// s = shared
+    /// p = private (copy on write)
+    perms: [char; 4],
+    /// Offset into the file (or "whatever").
+    offset: usize,
+    /// device (major, minor)
+    dev: (usize, usize),
+    /// inode on the device. 0 indicates that no inode is associated with the memory region (e.g. uninitalized data aka BSS).
+    inode: usize,
+    /// Usually the file backing the mapping.
+    ///
+    /// Note: The man page for proc includes a note about "coordination" by
+    /// using readelf to see the Offset field in ELF program headers. pnkfelix
+    /// is not yet sure if that is intended to be a comment on pathname, or what
+    /// form/purpose such coordination is meant to have.
+    ///
+    /// There are also some pseudo-paths:
+    /// "[stack]": The initial process's (aka main thread's) stack.
+    /// "[stack:<tid>]": a specific thread's stack. (This was only present for a limited range of Linux verisons; it was determined to be too expensive to provide.)
+    /// "[vdso]": Virtual dynamically linked shared object
+    /// "[heap]": The process's heap
+    ///
+    /// The pathname can be blank, which means it is an anonymous mapping
+    /// obtained via mmap.
+    ///
+    /// Newlines in pathname are replaced with an octal escape sequence.
+    ///
+    /// The pathname may have "(deleted)" appended onto it if the file-backed
+    /// path has been deleted.
+    ///
+    /// Note that modifications like the latter two indicated above imply that
+    /// in general the pathname may be ambiguous. (I.e. you cannot tell if the
+    /// denoted filename actually ended with the text "(deleted)", or if that
+    /// was added by the maps rendering.
+    pathname: OsString,
+}
+
+pub(super) fn parse_maps() -> Result<Vec<MapsEntry>, &'static str> {
```

Comment:
This logic only applies to linux I think. Netbsd has `/proc/curproc/`. macOS doesn't have `/proc` at all. As for other unixes I'm not quite sure.

Reply:
I'm happy to revise this so that only Linux pulls in the "interesting" `parse_maps` file, and every other target just goes to the no-op variant.

Reply:
I looked again, and realized that the sole calling module is solely defined here:

https://github.com/rust-lang/backtrace-rs/blob/8b83ba10fc8ccfc12ff75fcc3277286c0b9fad62/src/symbolize/gimli.rs#L175-L185


That makes this a lot easier to limit to just these small subset of targets.